community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -167,71 +160,186 @@ public Commitment append( TransactionRepresentation transaction, long expectedTr
                         "Received " + transaction + " with txId:" + expectedTransactionId +
                         " to be applied, but appending it ended up generating an unexpected txId:" + transactionId );
             }
-            return append0( transaction, transactionId );
+            return serializeToTransactionLog( transaction, transactionId );
         }
     }
 
-    private class TransactionCommitment implements Commitment
+    private static class TransactionCommitment implements Commitment
```

Comment:
What would you say is the difference between having a non-static internal class and a static internal w/ reference to parent?